File: LeetCode/April Challenge/3rd Day/Day_3.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int maximum=nums[0], currmax=nums[0];
        for(int i=1;i<nums.size();i++)
        {
            currmax=max(nums[i],currmax+nums[i]);
            maximum=max(maximum,currmax);
        }
        return maximum;
    }
};

File: LeetCode/April Challenge/3rd Day/Day_3.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        curr_max = nums[0]
        curr_min = nums[0]
        final_sum = nums[0]
        for i in range(1, len(nums)):
            temp = curr_max
            curr_max = max(max(curr_max + nums[i], curr_min + nums[i]), nums[i])
            curr_min = min(min(temp + nums[i], curr_min + nums[i]), nums[i])
            if curr_max > final_sum:
                final_sum = curr_max
        return final_sum
